app/models/manageiq/providers/openstack/cloud_manager/vm/associate_ip.rb

```diff
@@ -0,0 +1,47 @@
+module ManageIQ::Providers::Openstack::CloudManager::Vm::AssociateIp
+  extend ActiveSupport::Concern
+
+  included do
+    supports :associate_floating_ip do
+      if cloud_tenant.nil? || Rbac.filtered(cloud_tenant.floating_ips).empty?
```

Comment:
This is a question for @kbrock. He has been cleaning this out recently and giving some performance related usage tips. Ping @kbrock ?